lorawan/src/subnet.rs

```diff
@@ -0,0 +1,386 @@
+pub fn is_local_devaddr(devaddr: u32, netid_list: Vec<u32>) -> bool {
+    let netid = parse_netid(devaddr);
+    is_local_netid(netid, netid_list)
+}
+
+pub fn devaddr_from_subnet(subnetaddr: u32, netid_list: Vec<u32>) -> u32 {
+    let netid = subnet_addr_to_netid(subnetaddr, netid_list.clone());
+    let (lower, _upper) = netid_addr_range(netid, netid_list);
+    devaddr(netid, subnetaddr - lower)
+}
+
+pub fn subnet_from_devaddr(devaddr: u32, netid_list: Vec<u32>) -> u32 {
+    let netid = parse_netid(devaddr);
+    let (lower, _upper) = netid_addr_range(netid, netid_list);
+    lower + nwk_addr(devaddr)
+}
+
+fn netid_class(netid: u32) -> u32 {
+    netid >> 21
+}
+
+fn addr_len(netclass: u32) -> u32 {
```

Comment:
```rust
fn addr_len(netclass: u32) -> u32 {
    *[25, 24, 20, 17, 15, 13, 10, 7]
        .get(netclass as usize)
        .unwrap_or(&0)
}
```

Reply:
Hi Jay - This is my first attempt at Rust code.  You'll have to explain this line by line :-)

Reply:
No prob. We can pair if it would help

Reply:
I read up on the Rust raw pointer.  I think I understand now; this is useful to know.  Seems reasonable; updated.